openwebwork/webwork2
# OPL Statistics

This is a new feature which adds some basic statistical information to the Library Browser.  You can read about the details here: http://webwork.maa.org/wiki/OPL_Problem_Statistics but the short version is that problems in the library browser will look like
![opl_statistics](https://cloud.githubusercontent.com/assets/1774907/11903807/b280a1ec-a589-11e5-928b-817b6ca05493.png)

To get this running: 
1) Switch over to this branch. 
2) Download the latest version of the OPL
3) Run `update-OPL-statistics`. 

Things to test: 
- Test that you get both global and local statistics in the Library Browser. 
- Test that you can turn the feature on and off by using the two stats variables in `localOverrides.conf.dist`
- Check that formatting and help links are sane. 
- Check that if you have an open set, submissions do not change the local stats (after running `update-OPL-statistics` again) but if you change the due date to be closed then the local stats are updated.  

Note:  This also contains the `dump_past_answers` script which is being used in the other big data project.  It can be ignored for now.
